source/lac/trilinos_sparse_matrix.cc

```diff
@@ -2310,6 +2310,17 @@ namespace TrilinosWrappers
     return ((sizeof(TrilinosScalar)+sizeof(TrilinosWrappers::types::int_type))*
             matrix->NumMyNonzeros() + sizeof(int)*local_size() + static_memory);
   }
+
+  const MPI_Comm &SparseMatrix::get_mpi_communicator () const
+  {
+    static MPI_Comm communicator;
```

Comment:
I would prefer returning MPI_Comm by value so you don't have to create a static variable here. MPI_Comm is something like an int anyway.